src/gui/screen_menu_fw_update.cpp

```diff
@@ -59,12 +65,13 @@ void ScreenMenuFwUpdate::windowEvent(EventLock /*has private ctor*/, window_t *s
     if (event == GUI_event_t::CLICK) {
         MI_ON_RESTART *mi_restart = &Item<MI_ON_RESTART>();
         if (size_t(param) == 1) {
-            mi_restart->index = sys_fw_update_on_restart_is_enabled() ? 0 : 1;
-            mi_restart->Enable();
-        } else {
             mi_restart->Disable();
-            mi_restart->index = 0;
+            mi_restart->index = 1;
+        } else {
+            mi_restart->Enable();
+            mi_restart->index = sys_fw_update_on_restart_is_enabled() ? 0 : 1;
         }
+        Invalidate();
```

Comment:
Because I couldn't find member function Invalidate of the MI_ON_RESTART class, I used global one.